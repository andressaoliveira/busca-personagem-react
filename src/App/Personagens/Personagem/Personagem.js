import React, { Component } from "react"
import PropTypes from "prop-types"

import './Personagem.scss';

class Personagem extends Component {

  obterOcupacao = (ocupacoes) => {
    let ocupacao = "";
    for (let i = 0; i < ocupacoes.length; i++) {
      if (i !== 0)
        ocupacao = `${ocupacao}, `;
      ocupacao = `${ocupacao} ${ocupacoes[i]}`;
    }
    return ocupacao;
  }

  render() {
    const { personagem } = this.props;

    return (
      <div className="personagem" key={personagem.char_id}>
        <img src={personagem.img} width="100%" height="100%" alt={personagem.name} />
        <div className="dados">
          <div className="status">
            <div className="etiqueta" style={{ backgroundColor: personagem.status === 'Alive' ? '#008B45' : '#EE7600' }}>
              {personagem.status}
            </div>
          </div>
          <div className="info">
            <div className="nome">{personagem.name}</div>
            <div className="data-nascimento">â˜… {personagem.birthday}</div>
            <div className="ocupacao">{this.obterOcupacao(personagem.occupation)}</div>
          </div>
        </div>
      </div>
    )
  }
}

Personagem.propTypes = {
  personagem: PropTypes.object.isRequired,
}

export default Personagem
