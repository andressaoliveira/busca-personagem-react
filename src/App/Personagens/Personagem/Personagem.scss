.personagem{
  width: 250px;
  height: 350px;
  padding: 10px 5px;

  .dados{
    box-shadow: 0px -50px 36px #000 inset;
    width: 250px;
    height: 350px;
    position: relative;
    bottom: 353px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    .status{
      width: 100%;
      justify-content: flex-end;
      display: flex;
      .etiqueta{
        background-color: green;
        width: 100px;
        height: 30px;
        display: flex;
        justify-content: center;
        align-items: center;
        position: relative;
        right: 10px;
        top: 10px;
      }
    }
    .info{
      justify-content: flex-end;
      display: flex;
      flex-direction: column;
      padding-left: 10px;
      .nome{
        display: flex;
        font-size: 16px;
        font-weight: 500;
      }
      .data-nascimento{
        display: flex;
      }
      .ocupacao{
        display: flex;
        text-align: left;
      }
    }
  }
}
