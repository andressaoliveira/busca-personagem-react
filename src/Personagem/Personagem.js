import React, { Component } from "react"
import PropTypes from "prop-types"

import './Personagem.scss';

class Personagem extends Component {

  render() {
    const { personagem } = this.props;
    console.log(personagem)

    let imagem = personagem.image !== "" ? personagem.image : "https://i.pinimg.com/280x280_RS/46/87/d9/4687d9c8695cfb2d7acd2a303ae45228.jpg"

    return (
      <div className="personagem" key={personagem.id}>
        <img src={imagem} width="100%" height="100%" alt={personagem.name} />
        <div className="dados">
          <div className="status">
            <div className="etiqueta" style={{ backgroundColor: personagem.alive ? '#008B45' : '#EE7600' }}>
              {personagem.species}
            </div>
          </div>
          <div className="info">
            <div className="nome">{personagem.name}</div>
            <div className="data-nascimento">â˜… {personagem.dateOfBirth}</div>
            <div className="ocupacao">{personagem.house}</div>
          </div>
        </div>
      </div>
    )
  }
}

Personagem.propTypes = {
  personagem: PropTypes.object.isRequired,
}

export default Personagem
