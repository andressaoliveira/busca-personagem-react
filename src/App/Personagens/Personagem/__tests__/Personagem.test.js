import expect from "expect";
import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

import Personagem from "../index";

configure({ adapter: new Adapter() })

describe("Personagem", () => {
  function setup() {
    const props = {
      personagem: {
        char_id: 18,
        name: "Brandon Mayhew",
        birthday: "Unknown",
        occupation: [
          "Low-level meth distributer"
        ],
        img: "https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_badger-lg.jpg",
        status: "Alive",
        nickname: "Badger",
        appearance: [
          1,
          2,
          3,
          4,
          5
        ],
        portrayed: "Matt L. Jones"
      }
    };
    return shallow(<Personagem {...props} />);
  }

  const wrapper = setup();

  it("deve montar o componente sem deixar de funcionar", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("deve mostrar ocupacao com 1", () => {
    const ocupacao = ["ocupacao"];
    const resultado = wrapper.instance().obterOcupacao(ocupacao);

    expect(resultado).toBe(" ocupacao");
  });

  it("deve mostrar ocupacao com mais de 1", () => {
    const ocupacao = ["ocupacao 1", "ocupacao 2"]
    const resultado = wrapper.instance().obterOcupacao(ocupacao);

    expect(resultado).toBe(" ocupacao 1,  ocupacao 2");
  });
});
