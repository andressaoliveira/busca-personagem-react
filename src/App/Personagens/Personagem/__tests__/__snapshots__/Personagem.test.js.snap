// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personagem deve montar o componente sem deixar de funcionar 1`] = `ShallowWrapper {}`;
